spec/integration/test_runner_spec.rb

```diff
@@ -1,133 +1,141 @@
-require "rails_helper"
-Dir[Rails.root.join("lib/integration_helpers/**/*.rb")].sort.each { |f| require f }
-
-##### NOTE #####
-#
-# This spec can be run for just one worksheet, or with varying levels of verbosity with the
-# executable bin/ispec, for example:
-#
-#     bin/ispec -r -vv -w NPE6-1
-#
-# Will force a refresh of all the spreadsheets, process only worksheet NPE6-1, and have verbosity level 2
-# (show details of all payloads and responses)
-#
-#    bin/ispec -h # show help text
-#
-RSpec.describe "IntegrationTests::TestRunner", type: :request do
-  let(:spreadsheet_title) { "CFE Integration Test V3" }
-  let(:target_worksheet) { ENV["TARGET_WORKSHEET"] }
-  let(:verbosity_level) { (ENV["VERBOSE"] || "0").to_i }
-  let(:refresh) { (ENV["REFRESH"] || "false") }
-
-  let(:spreadsheet_file) { Rails.root.join("tmp/integration_test_data.xlsx") }
-  let(:spreadsheet) { Roo::Spreadsheet.open(spreadsheet_file.to_s) }
-  let(:worksheet_names) { spreadsheet.sheets }
-  let(:headers) { { "CONTENT_TYPE" => "application/json", "Accept" => "application/json;version=5" } }
-
-  before do
-    setup_test_data
-    mock_lfa_responses
-  end
-
-  describe "run integration_tests", :vcr do
-    ispec_run = ENV["ISPEC_RUN"].present?
-
-    if ispec_run
-      it "processes all the tests on all the sheets" do
-        failing_tests = []
-        test_count = 0
-        group_runner = TestCase::GroupRunner.new(verbosity_level, refresh)
-        group_runner.each do |worksheet|
-          next if target_worksheet.nil? && worksheet.skippable?
-          next if target_worksheet.present? && target_worksheet != worksheet.worksheet_name
-
-          test_count += 1
-          puts ">>> RUNNING TEST #{worksheet.description} <<<".yellow unless silent?
-          pass = run_test_case(worksheet)
-          failing_tests << worksheet.description unless pass
-          result_message(failing_tests, test_count) unless silent?
-        end
-        expect(failing_tests).to be_empty, "Failing tests: #{failing_tests.join(', ')}"
-      end
-    else
```

Comment:
How about we make this line `elsif ENV["PRIVATE_KEY_ID"].present?` ? That way these tests are still runnable if you do happen to have credentials, we just also have a way of having CI work without those credentials.

Reply:
great solution, will do